std/assembly/string.ts

```diff
@@ -484,6 +486,114 @@ import { idof } from "./builtins";
     // releases result
   }
 
+  toLowerCase(): String {
+    var len = <usize>this.length;
+    if (!len) return this;
+    var codes = __alloc(len * 2 * 2, idof<String>());
+    var j: usize = 0;
+    for (let i: usize = 0; i < len; ++i, ++j) {
+      let c = <u32>load<u16>(changetype<usize>(this) + (i << 1));
+      if (isAscii(c)) {
+        store<u16>(codes + (j << 1), toLower8(c));
+      } else {
+        if ((c - 0xD7FF < 0xDC00 - 0xD7FF) && i < len - 1) {
```

Comment:
Can you add some comments here before the respective ifs, like which special cases are covered where? I don't for example understand the monkey patch.

Reply:
Ok, it;'s not monkey patch, just check and read second part of surrogate pair

Reply:
Fixed